reposerver/repository/repository.go

```diff
@@ -130,22 +144,24 @@ func (s *Service) runRepoOperation(
 		}
 	}
 
-	if !settings.noCache && getCached(revision) {
+	if !isResolvedRevision && !settings.noCache && getCached(revision) {
 		return nil
 	}
 
 	s.metricsServer.IncPendingRepoRequest(repo.Repo)
 	defer s.metricsServer.DecPendingRepoRequest(repo.Repo)
 
 	if settings.sem != nil {
-		err = settings.sem.Acquire(c, 1)
+		err = settings.sem.Acquire(ctx, 1)
 		if err != nil {
 			return err
 		}
 		defer settings.sem.Release(1)
 	}
 
 	if source.IsHelm() {
+		s.repoLock.Lock(source.Chart)
```

Comment:
missing lock

Reply:
Lock is required only for git. Helm chart is exported into temp directory every time and does not require locking

Reply:
you're quite right, i noted this after and did not revert